### PR TITLE
Fix case on dell warranty check condition

### PR DESCRIPTION
**Pull request by mjrasobarnett**
Hi, I just tried this out on some Dell servers we have and found I needed to fix the case on this check for it to work. Great work btw, thanks for this!

### CODE REVIEW

**Comment by jhaals** [comment]
Hi @mjrasobarnett! I haven't done work on this puppet module in a long time, fun to see that it's in use! Is it working for you guys when it's lowercase?

**Comment by mjrasobarnett** [comment]
Hi, no worries, yes I needed to change this to lowercase for it to work as you are transforming the 'manufacturer' fact to lower case before doing the comparison.

I should have put this on a different branch before sending in as I can see another commit I've made (63722dc) is showing up on this PR, however you might want to include that too as it fixes a different problem -- when looping over all the warranties the dell api returns, I've changed it to select the warranty date furthest in the future, i.e: the expiry date of the final warranty. As it stands it will just return the one that the dell api returns last, which in my case was the earliest warranty period.

I'm not very experienced with ruby though so there is probably a nicer way to do all this. We have a lot of dell kit though so this has been a nice little addition :)

File: lib/facter/warranty.rb [hunk 1]
```diff
@@ -28,16 +28,20 @@ def create_dell_warranty_cache(cache)
...
+      if expiration_date < end_date
         warranty = true
-        expiration_date = end_date.strftime("%Y-%m-%d")
+        expiration_date = end_date
       end
   end
   rescue
   end
 
+  if expiration_date > Time.now()
```
Comment (jhaals) [inline lib/facter/warranty.rb]: this can be reduced to one line `warranty = true if expiration_date > Time.now()`

**Comment by jhaals** [comment]
Glad you're using and improving this module. No worries, it can be done in the same PR.

**Pull request merged**